halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/canary/EditCanaryCommand.java

```diff
@@ -0,0 +1,132 @@
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.halyard.cli.command.v1.config.canary;
+
+import com.beust.jcommander.Parameter;
+import com.beust.jcommander.Parameters;
+import com.netflix.spinnaker.halyard.cli.command.v1.config.AbstractConfigCommand;
+import com.netflix.spinnaker.halyard.cli.services.v1.Daemon;
+import com.netflix.spinnaker.halyard.cli.services.v1.OperationHandler;
+import com.netflix.spinnaker.halyard.cli.ui.v1.AnsiUi;
+import com.netflix.spinnaker.halyard.config.model.v1.canary.Canary;
+import lombok.AccessLevel;
+import lombok.Getter;
+
+@Parameters(separators = "=")
+public class EditCanaryCommand extends AbstractConfigCommand {
+  @Getter(AccessLevel.PUBLIC)
+  private String commandName = "edit";
+
+  @Getter(AccessLevel.PUBLIC)
+  private String description = "Edit Spinnaker's canary analysis settings.";
+
+  @Parameter(
+      names = "--redux-logger-enabled",
+      arity = 1,
+      description = "Whether or not to enable redux logging in the canary module in deck."
+  )
+  private Boolean reduxLoggerEnabled = true;
+
+  @Parameter(
+      names = "--default-metrics-account",
+      arity = 1,
+      description = "Name of metrics account to use by default."
+  )
+  private String defaultMetricsAccount = null;
+
+  @Parameter(
+      names = "--default-storage-account",
+      arity = 1,
+      description = "Name of storage account to use by default."
+  )
+  private String defaultStorageAccount = null;
+
+  @Parameter(
+      names = "--default-judge",
+      arity = 1,
+      description = "Name of canary judge to use by default (e.g. NetflixACAJudge-v1.0)."
+  )
+  private String defaultJudge = null;
+
+  @Parameter(
+      names = "--default-metrics-store",
+      arity = 1,
+      description = "Name of metrics store to use by default (e.g. atlas, datadog, prometheus, stackdriver)."
+  )
+  private String defaultMetricsStore = null;
+
+  @Parameter(
+      names = "--stages-enabled",
+      arity = 1,
+      description = "Whether or not to enable canary stages in deck."
+  )
+  private Boolean stagesEnabled = true;
+
+  @Parameter(
+      names = "--atlasWebComponentsUrl",
+      arity = 1,
+      description = "Location of web components to use for Atlas metric configuration."
+  )
+  private String atlasWebComponentsUrl = null;
+
+  @Parameter(
+      names = "--templates-enabled",
+      arity = 1,
+      description = "Whether or not to enable custom filter templates for canary configs in deck."
+  )
+  private Boolean templatesEnabled = true;
```

Comment:
Since this is an "edit" command, if this flag isn't supplied by a user it will always be set to `true`. I would drop these defaults.

Reply:
I want it to default to `true`. Is there a better way to accomplish that?

Reply:
Fixed as per offline discussion.